File: internal/content/testdata/bench/go/nc-trapping-rain-water.go
func trap(heights []int) int {
	if len(heights) == 0 {
		return 0
	}
	i, j := 0, len(heights)-1
	leftMax, rightMax := heights[i], heights[j]
	total := 0
	for i < j {
		if leftMax < rightMax {
			i++
			if heights[i] > leftMax {
				leftMax = heights[i]
			}
			total += leftMax - heights[i]
		} else {
			j--
			if heights[j] > rightMax {
				rightMax = heights[j]
			}
			total += rightMax - heights[j]
		}
	}
	return total
}

File: internal/content/testdata/bench/csharp/nc-trapping-rain-water.cs
public class Solution {
    public long trap(long[] heights) {
        if (heights.Length == 0) return 0;
        int i = 0, j = heights.Length - 1;
        long leftMax = heights[i], rightMax = heights[j];
        long total = 0;
        while (i < j) {
            if (leftMax < rightMax) {
                i++;
                leftMax = Math.Max(leftMax, heights[i]);
                total += leftMax - heights[i];
            } else {
                j--;
                rightMax = Math.Max(rightMax, heights[j]);
                total += rightMax - heights[j];
            }
        }
        return total;
    }
}
